src/wallet/api/wallet.cpp

```diff
@@ -748,4 +742,27 @@ void WalletImpl::pauseRefresh()
 }
 
 
+bool WalletImpl::isNewWallet() const
+{
+    // in case wallet created without daemon connection, closed and opened again,
+    // it's the same case as if it created from scratch, i.e. we need "fast sync"
+    // with the daemon (pull hashes instead of pull blocks)
+    return !(blockChainHeight() > 1);
+}
+
+void WalletImpl::doInit(const string &daemon_address, uint64_t upper_transaction_size_limit)
+{
+    m_wallet->init(daemon_address, upper_transaction_size_limit);
+
+    // in case new wallet, this will force fast-refresh (pulling hashes instead of blocks)
+    if (isNewWallet()) {
+        m_wallet->set_refresh_from_block_height(daemonBlockChainHeight());
+    }
```

Comment:
That looks like it could match restoring from seed (and thus restoring from seed would miss previous payments). Have you tested that ?